src/workerd/api/sql.c++

```diff
@@ -18,6 +18,24 @@ jsg::Ref<SqlStorage::Cursor> SqlStorage::exec(jsg::Lock& js, kj::String querySql
   return jsg::alloc<Cursor>(*sqlite, regulator, querySql, kj::mv(bindings));
 }
 
+SqlStorage::SavepointResult SqlStorage::savepoint(jsg::Lock& js, jsg::Function<jsg::Value()> callback) {
+  // SAVEPOINT is a readonly statement, but we need to trigger an outer TRANSACTION
+  KJ_IF_MAYBE(cb, sqlite->onWriteCallback) {
+    (*cb)();
+  }
+  // TODO(sqlite) allow nested savepoints, using a new savepoint name for each depth
+  sqlite->prepare("SAVEPOINT aaa;").run();
```

Comment:
We should prefix the savepoint name with `_cf_` just to be consistent that all our internal names that we give to sqlite are so prefixed.